apps/coscrad-frontend/src/components/resources/terms/term-search-bar.tsx

```diff
@@ -12,15 +12,116 @@ import {
     TextField,
 } from '@mui/material';
 import { useContext, useState } from 'react';
+import { useAppDispatch } from '../../../app/hooks';
 import { ConfigurableContentContext } from '../../../configurable-front-matter/configurable-content-provider';
-import { ALL_PROPERTIES_SEARCH_KEY } from '../../../store/slices/resources';
+import {
+    ALL_PROPERTIES_SEARCH_KEY,
+    fetchTerms,
+    filterTermsInMemory,
+    IndexSearchScope,
+    IUserDefinedFilter,
+    setTermFilters,
+    useLoadableTerms,
+} from '../../../store/slices/resources';
 
-interface SearchBarProps {
-    onValueChange: (
-        scope: keyof ITermViewModel | typeof ALL_PROPERTIES_SEARCH_KEY,
-        newValue: string
-    ) => void;
+const compileMultilingualTextContainsQuery = (
```

Comment:
We should break this out into a separate file now.

Reply:
Actually, we should wait. It needs to be generalized for other resource types and possibly even broken out to the server.